### ISSUE TITLE
[20200229] - Local Environment in Window

### ISSUE THREAD

**Issue by noklam** [fastai/fastpages#144]
#28  Please include the following information
I get these error for
```
docker-compose down || true;
Removing fastpages_converter_1 ... done
Removing fastpages_fastpages_1 ... done
Removing fastpages_watcher_1   ... done
Removing fastpages_jekyll_1    ... done
Removing fastpages_jupyter_1   ... done
Removing network fastpages_default
docker-compose up
Creating network "fastpages_default" with the default driver
Creating fastpages_jekyll_1    ... done
Creating fastpages_watcher_1   ... done
Creating fastpages_jupyter_1   ... done
Creating fastpages_converter_1 ... done
Creating fastpages_fastpages_1 ... done
Attaching to fastpages_converter_1, fastpages_jupyter_1, fastpages_watcher_1, fastpages_fastpages_1, fastpages_jekyll_1
converter_1  | standard_init_linux.go:211: exec user process caused "no such file or directory"
fastpages_1  | standard_init_linux.go:211: exec user process caused "no such file or directory"

```
Then I find out that multiple links are link to /fastpages/...
Which caused the problem as this does not match my directory.

My current solution is to 

1. Steps to reproduce the problem
Window 10
`make server`
2. A link to your repository where error is occuring
The Two dockers cannot start successfully
 standard_init_linux.go:211: exec user process caused "no such file or directory
![image](https://user-images.githubusercontent.com/18221871/75606816-a220c780-5b2b-11ea-9569-a6acf83ee5da.png)

**Comment by noklam** [comment]
I does some more digging, which I found notebook post are not generated probably. (As localhost:4000 show only markdown post but not notebooks).

I can reproduce this error message by manually edit notebook to force it regenerate notebook post.

jekyll_1     |     Server address: http://0.0.0.0:4000/
jekyll_1     |   Server running... press ctrl-c to stop.


watcher_1    | /bin/sh: 1: /fastpages/action_entrypoint.sh: not found
watcher_1    | /bin/sh: 1: /fastpages/action_entrypoint.sh: not found

I then try to login to watcher_1 docker to check if this file exist, it does.

![image](https://user-images.githubusercontent.com/18221871/75607153-14df7200-5b2f-11ea-9d02-d3b128557de5.png)

Then I did some more digging by manually execute the script
![image](https://user-images.githubusercontent.com/18221871/75607195-6425a280-5b2f-11ea-90b2-515dba1d68c3.png)

It seems to related to some special token with issue editing in Windows. Therefore, I change to save from "CRLF" to "LF" but got no luck yet. And it complains about something called `directory2post.py` which I am not sure where is this file (can't find in fastpages nor nbdev)
https://stackoverflow.com/questions/51888625/docker-r-command-not-found-on-windows

![image](https://user-images.githubusercontent.com/18221871/75607340-0003de00-5b31-11ea-8d95-55e29b201659.png)

**Comment by noklam** [comment]
Maybe I am not cleaning the token correctly, I clean the token in my own directory, I am not sure if the files in Docker are pulled from the directory or somewhere else.

**Comment by noklam** [comment]
It seems that I have to clean up in watcher_1 docker directly with commands
```
sed -i 's/\r//g' /fastpages/action_entrypoint.sh
sed -i 's/\r//g' /fastpages/word2post.sh

```
I can successfully to generate notebook post now!

![image](https://user-images.githubusercontent.com/18221871/75607460-5a516e80-5b32-11ea-8911-3647df112627.png)

But I can't create a fix PR from my side as I don't know where does the file in /fastpages/*.sh come from. I tried clean up on my local repository and re-run `make server` but does not work, it only works when I clean up in docker manually.

**Comment by hamelsmu** [comment]
I think we need to focus on reproducing this error

```
converter_1  | standard_init_linux.go:211: exec user process caused "no such file or directory"
fastpages_1  | standard_init_linux.go:211: exec user process caused "no such file or directory"
```

If you clone this repo fresh and run `make server`  Do you get this error?

**Comment by hamelsmu** [comment]
I wonder if I can add a test in Windows to the CI.  I think I can

**Comment by noklam** [comment]
For the record I am running this in Window 10 with docker, I tried WSL with no luck.
Here is how I reproduce
```
git clone https://github.com/fastai/fastpages.git
cd fastpages
make build
make server
```

![image](https://user-images.githubusercontent.com/18221871/75611441-8a5e3900-5b55-11ea-9713-d3f58bc9cdc9.png)

**Comment by hamelsmu** [comment]
Thanks @noklam this is very helpful.  We will keep this issue open until we figure this out.  Thanks for bringing this to our attention

**Comment by hamelsmu** [comment]
@noklam I'll go ahead and close this issue since I merged your PR.  Please feel free to reopen this issue if the problem persists.

Thank you

**Comment by noklam** [comment]
Thanks @hamelsmu

**Comment by xnutsive** [comment]
I was down with a cold most of the week ;( Thank you @noklam for reporting and fixing it!

**Comment by noklam** [comment]
@xnutsive thx! 😃

**Comment by arisbw** [comment]
Hello, I still have an issue with this one. I tried to build a local environment based on my repo (https://github.com/arisbw/p/) but didn't succeed. And also I tried to make a local server based on the fresh repo of fastpages (ref: https://github.com/fastai/fastpages/issues/144#issuecomment-592964219) but didn't find any success either.

Instead of using https://127.0.0.1:4000 as the preview to my page, the server address was directed to http://0.0.0.0:4000/

I'm using WSL2 with docker installed.

Here is the screenshot when I tried to run `make server` based on the original repo of fastpages:

![image](https://user-images.githubusercontent.com/4155427/97629843-495e5180-1a61-11eb-81bd-4ca5aeff7610.png)
![image](https://user-images.githubusercontent.com/4155427/97629886-58dd9a80-1a61-11eb-87d0-89ad07717338.png)

**Comment by hamelsmu** [comment]
@arisbw what's wrong though?  I don't see anything that is broken?  You can go to that URL and see fastpages

**Comment by arisbw** [comment]
Sorry that I didn't make it clear. I did try to access though to the server address: http://0.0.0.0:4000/ but I didn't see the site.

Here is the screenshot when I attempted to access the address:
![image](https://user-images.githubusercontent.com/4155427/97717156-0f3f8f00-1af7-11eb-80e4-5f70c809bb57.png)

**Comment by hamelsmu** [comment]
Not sure what’s wrong it’s perhaps your local networking or Docker settings

**Comment by ClaudiuFilip110** [comment]
Hello, I'm quite a noob at networking and I had a similar problem as @arisbw. From what I understand the server runs locally so it automatically finds the address(localhost). 

I fixed the problem by addressing another issue. Docker couldn't find the correct base url so I commented line 37 in the _config.yml 
![image](https://user-images.githubusercontent.com/44347749/98406982-8c2eb380-2077-11eb-8089-cdbd890a20b4.png)
```
baseurl: "/Blogging" # the subpath of your site, e.g. "/blog".
```
This fixed both issues for me.
![image](https://user-images.githubusercontent.com/44347749/98407027-9ea8ed00-2077-11eb-8341-c3bf0e87efff.png)

I hope this helps.
